added rum support. fixed compile errors in ratom. (fixes issue #27)

bamarco opened the pull request.

bamarco committed:
added rum support. fixed compile errors in ratom

bamarco committed:
minor bugfix

bamarco committed:
minor bugfix

alexandergunnarson commented on src/posh/lib/ratom.cljc:
```diff
@@ -17,146 +17,147 @@
-  (if-cljs &env `(array ~@args)
-                `(doto (ArrayList.) ~@(for [arg args] `(.add ~arg))))))
+   (defmacro array-list [& args]
+     (if-cljs &env `(into [] ~@args)
```
This is going to cause issues with ‘alength’  and ‘aget’ further down, as well as a few other things. A JS array is extensible so it’s basically an arraylist. Persistent vectors’ properties are too dissimilar to just swap it in here.

bamarco replied:
The array macro doesn't seem to exist in cljs anymore. What is the correct function to use then?

alexandergunnarson replied:
Sorry, don't know why I didn't respond to this! Just wanted to say that `array` is definitely still there — see also https://github.com/clojure/clojurescript/blob/master/src/main/clojure/cljs/core.cljc#L2482.

alexandergunnarson replied:
Though I think I might know why it's not compiling properly — try:
```
`(cljs.core/array ~@args)
```

rather than
```
`(array ~@args)
```

Because of how CLJS-in-CLJ compilation works, possibly the `array-list` macro was resolving `array` to `clojure.core/array` rather than `cljs.core/array`. Not sure — I'd have to test it out.

bamarco committed:
:derive-reaction added to make rum and reagent compatible

bamarco committed:
turn off prn statements

bamarco committed:
better replacement for array

bamarco committed:
fully named cljs.core/array which actually fixes the issue

mpdairy commented:
Nice, is this ready to merge then? I don't have clojure set up on this computer and didn't write any tests anyways so I can't test it.

ojw commented:
Hey what's the status of this PR?  Is there anything an outsider can do to help?  Rum support in Posh would be great.

metasoarous commented:
@ojw I'll defer to @bamarco and @alexandergunnarson on this question, but to my recollection things are more or less ready to go and just need to be tested out. If you're comfortable with Rum, it might be super helpful to have you test things out in coordination with Brian and Alexander.

alexandergunnarson commented on src/posh/plugin_base.cljc:
```diff
@@ -99,19 +99,20 @@
-                                                       :ratoms (dissoc (:ratoms posh-atom-val) storage-key)
-                                                       :reactions (dissoc (:reactions posh-atom-val) storage-key)))))))]
+                 query-reaction       ((:derive-reaction dcfg) [query-ratom] storage-key (fn [q] q))]
+                ;  query-reaction       ((:make-reaction dcfg)
```
I haven't thought through what would happen if you change the implementation of `query-reaction` to what you have now, but assuming your version works, could you @bamarco remove the commented portion of code?

alexandergunnarson commented on src/posh/rum.cljc:
```diff
@@ -0,0 +1,31 @@
+    [datascript.core :as d]))
+
+(defn make-reaction [f & {:as local-mixin :keys [on-dispose]}]
+  ; (prn "make-reaction" (str f))
```
Looks like this is a lingering todo @bamarco?

alexandergunnarson commented on src/posh/rum.cljc:
```diff
@@ -0,0 +1,31 @@
+  ;; TODO: handle at least the on-dispose method of the local mixin
+  (rum/derived-atom [] ::no-refs-anyways f))
+
+; (defn make-reaction2 [f & {:as local-mixin :keys [on-dispose]}]
```
Is this commented code block okay to be removed @bamarco?

alexandergunnarson commented:
@ojw @metasoarous @bamarco I haven't tested out any part of this PR yet but in theory yes, it should be ready to go once we figure out some of the concerns I posted in the review comments. @ojw I think it would be a big help to add some tests! Since the repo appears to have none, then even one good one would help :) Thanks!

denistakeda committed:
Use the new project name for clojars

- Use the new project name
- Fix readme

denistakeda committed:
Merge pull request #1 from denistakeda/update-dependencies

Use the new project name for clojars

denistakeda committed:
Make the reagent namespace testable in Clojure

- Rename `reagent.cljs` to `reagent.cljc`
- Write simple test to verify this works
- Don't use unported defs in Clojure environment

denistakeda committed:
Merge pull request #2 from denistakeda/test-in-clojure

Make the reagent namespace testable in Clojure

denistakeda committed:
Introduce Kaocha testing framework

- Introduce Kaocha
- Fix some tests without assertions

denistakeda committed:
Add prepush git hook to not push with broken tests

denistakeda committed:
Merge pull request #3 from denistakeda/introduce.kaocha

Add kaocha testing frameworks and pre-push hooks

bamarco committed:
merge conflicts

bamarco committed:
on-dispose no longer broken for reagent plugin to accomodate rum plugin

bamarco committed:
merged with denistakeda

bamarco committed:
fixed indentation

bamarco commented:
Moving the pr to the new repo

bamarco commented:
@ojw finally finished the rum plugin. it is pending in the new posh repository
